ros/src/computing/perception/detection/packages/lidar_tracker/nodes/imm_ukf_pda_tracker/src/imm_ukf_pda_tracker.cpp

```diff
@@ -0,0 +1,858 @@
+//                     double cross1 = getIntersectCoef(vec1x, vec1y, vec2x, vec2y, px, py, cp1x, cp1y);
+//                     double cross2 = getIntersectCoef(vec1x, vec1y, vec3x, vec3y, px, py, cp1x, cp1y);
+//                     double cross3 = getIntersectCoef(vec3x, vec3y, vec2x, vec2y, px, py, cp1x, cp1y);
+//                     // check if px, py are in triangle vec1, vec3, vec4
+//                     double cross4 = getIntersectCoef(vec1x, vec1y, vec4x, vec4y, px, py, cp2x, cp2y);
+//                     double cross5 = getIntersectCoef(vec1x, vec1y, vec3x, vec3y, px, py, cp2x, cp2y);
+//                     double cross6 = getIntersectCoef(vec3x, vec3y, vec4x, vec4y, px, py, cp2x, cp2y);
+//                     // check if inside or not
+//                     if((cross1 > 0 && cross2>0&&cross3>0)||(cross4>0 && cross5 > 0 && cross6>0)){
+//                         trackNumVec_[i] = 5;
+//                         trackNumVec_[j] = 0;
+//                     }
+//                 }
+//             }
+//         }
+//     }
+// }
+
+void ImmUkfPda::updateLabel(UKF target, autoware_msgs::DetectedObject& dd)
+{
+  int tracking_num = target.tracking_num_;
+  // cout << "trackingnum "<< trackingNum << endl;
+  if (target.is_static_)
+  {
+    dd.label = "Static";
+  }
+  else if (tracking_num > TrackingState::Die && tracking_num < TrackingState::Stable)
+  {
+    dd.label = "Initialized";
+  }
+  else if (tracking_num == TrackingState::Stable)
+  {
+    dd.label = "Stable";
+  }
+  else if (tracking_num > TrackingState::Stable && tracking_num <= TrackingState::Lost)
+  {
+    dd.label = "Lost";
+  }
+  else
+  {
+    dd.label = "None";
+  }
+}
+
+void ImmUkfPda::updateJskLabel(UKF target, jsk_recognition_msgs::BoundingBox& bb)
+{
+  int tracking_num = target.tracking_num_;
+  // cout << "trackingnum "<< trackingNum << endl;
+  if (target.is_static_)
+  {
+    bb.label = 15;  // white color
+  }
+  else if (tracking_num == TrackingState::Stable)
+  {
+    bb.label = 2;  // orange color
+  }
+}
+
+bool ImmUkfPda::isVisible(UKF target)
+{
+  bool is_visible = false;
+  int tracking_num = target.tracking_num_;
+  if (tracking_num == TrackingState::Stable || target.is_static_)
+  {
+    is_visible = true;
+  }
+  else
+  {
+    is_visible = false;
+  }
+  return is_visible;
+}
+
+void ImmUkfPda::initTracker(autoware_msgs::CloudClusterArray input, double timestamp)
+{
+  for (size_t i = 0; i < input.clusters.size(); i++)
+  {
+    double px = input.clusters[i].bounding_box.pose.position.x;
+    double py = input.clusters[i].bounding_box.pose.position.y;
+    Eigen::VectorXd init_meas = Eigen::VectorXd(2);
+    init_meas << px, py;
+
+    UKF ukf;
+    ukf.initialize(init_meas, timestamp);
+    targets_.push_back(ukf);
+  }
+  timestamp_ = timestamp;
+  init_ = true;
+  return;
+}
+
+void ImmUkfPda::secondInit(double dt, std::vector<autoware_msgs::CloudCluster> cluster_vec, UKF& target)
```

Comment:
`cluster_vec` does not modified. 
Please use const + reference/pointer to exclude copies.